hack/dockerfiles/archutil.Dockerfile

```diff
@@ -11,49 +16,47 @@ RUN apt-get update && apt-get --no-install-recommends install -y \
   binutils-mips64-linux-gnuabi64
 WORKDIR /src
 
-
 FROM base AS exit-amd64
-COPY fixtures/exit.amd64.S .
+COPY util/archutil/fixtures/exit.amd64.S .
 RUN x86_64-linux-gnu-gcc -static -nostdlib -o exit exit.amd64.S
```

Comment:
Is it missing `--noexecstack` that is causing the validation issue?

Seems to contain a temp filename.

```
» diff d1.dis d2.dis                                                                                                                                                !4185
2c2
< f1.bin:	file format elf64-x86-64
---
> f2.bin:	file format elf64-x86-64
25c25
<   400107: 7f c6                        	jg	0x4000cf </tmp/cc2EicSI.o+0x4000cf>
---
>   400107: 7f c6                        	jg	0x4000cf </tmp/ccxc7wrM.o+0x4000cf>
146,148c146,151
<        6: 63 63 32                     	movslq	50(%rbx), %esp
<        9: 45 69 63 53 49 2e 6f 00      	imull	$7286345, 83(%r11), %r12d # imm = 0x6F2E49
<       11: 76 31                        	jbe	0x44 <.symtab+0x44>
---
>        6: 63 63 78                     	movslq	120(%rbx), %esp
>        9: 63 37                        	movslq	(%rdi), %esi
>        b: 77 72                        	ja	0x7f <.symtab+0x7f>
>        d: 4d 2e                        	cs
>        f: 6f                           	outsl	(%rsi), %dx
>       10: 00 76 31                     	addb	%dh, 49(%rsi)


» diff f1.xxd f2.xxd                                                                                                                                                !4186
141c141
< 00001180: 3245 6963 5349 2e6f 0076 3100 7632 0065 7869 7400 5f5f 6273 735f 7374 6172 7400  2EicSI.o.v1.v2.exit.__bss_start.
---
> 00001180: 7863 3777 724d 2e6f 0076 3100 7632 0065 7869 7400 5f5f 6273 735f 7374 6172 7400  xc7wrM.o.v1.v2.exit.__bss_start.
```

Reply:
> Is it missing `--noexecstack` that is causing the validation issue?

Doesn't look like it: https://github.com/moby/buildkit/actions/runs/7752978105/job/21143440422#step:4:724

Reply:
I can't see how the compiler has interpreted and potentially modified the original assembly instructions with temp files like `jg	0x4000cf </tmp/cc2EicSI.o+0x4000cf>`, even with debug output `-S -v` and `-fverbose-asm`

And striping debug info and ident directives doesn't seem to solve this. I think optimization is already disabled with gcc by default so should not be this.

Reply:
What is the the difference now in binary? There is still `/tmp/_.o` in the binary, even with no execstack?